app/__init__.py

```diff
@@ -48,12 +48,62 @@ def create_app(config_class, database_uri=None):
         ]
     )
 
-    # Set content security policy
+    SELF = "'self'"
+
     csp = {
-        "default-src": f"'self' {app.config['FLASKS3_CDN_DOMAIN']}",
-        "script-src": f"'self' {app.config['FLASKS3_CDN_DOMAIN']}"
-        + " 'sha256-GUQ5ad8JK5KmEWmROf3LZd9ge94daqNvd8xy9YS1iDw='"  # pragma: allowlist secret
-        + " 'sha256-l1eTVSK8DTnK8+yloud7wZUqFrI0atVo6VlC6PJvYaQ='",  # pragma: allowlist secret
+        "default-src": f"'self' {app.config['FLASKS3_CDN_DOMAIN']} ",
+        "script-src": (
+            [
+                SELF,
+                f"{app.config['FLASKS3_CDN_DOMAIN']}",
+                "https://cdn.jsdelivr.net/npm/universalviewer@4.0.25/dist/umd/5315.4c3c820c7f8b3cc26be6.js",
+                "https://cdn.jsdelivr.net/npm/universalviewer@4.0.25/dist/umd/998.9c1bd6b181b8236d95c2.js",
+                "https://cdn.jsdelivr.net/npm/universalviewer@4.0.25/dist/umd/7484.468f27df41f99efd4b79.js",
+                "https://cdn.jsdelivr.net/npm/universalviewer@4.0.25/dist/umd/2185.34b4770909a62ebe892e.js",
+                "https://cdn.jsdelivr.net/npm/universalviewer@4.0.25/dist/umd/3171.f31082f3b568ce907389.js",
+                "https://cdn.jsdelivr.net/npm/universalviewer@4.0.25/dist/umd/4864.b0b319b4f29542847e0e.js",
+                "https://cdnjs.cloudflare.com/ajax/libs/pdf.js/3.11.174/pdf.min.js",
+                "https://cdnjs.cloudflare.com/ajax/libs/pdf.js/3.11.174/pdf.worker.min.js",
+                "'sha256-GUQ5ad8JK5KmEWmROf3LZd9ge94daqNvd8xy9YS1iDw='",  # pragma: allowlist secret
+                "'sha256-l1eTVSK8DTnK8+yloud7wZUqFrI0atVo6VlC6PJvYaQ='",  # pragma: allowlist secret
+            ]
+        ),
+        "style-src": [
+            SELF,
+            "'sha256-aqNNdDLnnrDOnTNdkJpYlAxKVJtLt9CtFLklmInuUAE='",  # pragma: allowlist secret
```

Comment:
How easy are these going to be to maintain?

Is this relevant or have we tested it? [Firefox does not support hash-based CSP](https://bugzilla.mozilla.org/show_bug.cgi?id=1409200)

Reply:
These hashes should be the same and shouldn't really change going forward as they are the hash of each style required by UV. We're also not using strict-dynamic in our CSP and I've done some testing in Firefox where the PDF is loaded correctly.

Reply:
So everything is good as long as we continue to use the same version of UV?

Reply:
Yes that's correct. So the CSP hashes in either script-src or style-src are to allow certain styles or inline scripts. See: https://content-security-policy.com/hash/